### PR TITLE
Sync X-Delete-At header

### PR DESCRIPTION
**Pull request by thiagodasilva**
Added a new option to propagate the expiration header (Default: false)
When enabled, the `X-Delete-At` header is synced to remote location.
This option is only valid when syncing to a Swift target as S3 does not
support object expiration at the object level, only at the bucket level.
Furthermore, the new option ignored if `remote_delete_after` is set

### CODE REVIEW

**Comment by thiagodasilva** [comment]
> I really don't like the name 'allowed_headers'. I think something like 'propagated_headers' makes more sense. Other than that, I think it is good.

@mdegerne good idea, i'll make the change.

File: test/unit/test_sync_swift.py [hunk 1]
```diff
@@ -326,6 +327,38 @@ def test_meta_unicode(self, mock_swift):
                 'x-object-meta-new': '\xf0\x9f\x91\x8d',
                 'x-object-meta-old': 'updated'})
 
+    @mock.patch('s3_sync.sync_swift.swiftclient.client.Connection')
+    def test_propagate_expiration(self, mock_swift):
+        self.mapping['propagate_expiration'] = 'true'
```
Comment (timuralp) [inline test/unit/test_sync_swift.py]: And use a boolean here.

File: s3_sync/sync_swift.py [hunk 2]
```diff
@@ -14,17 +14,19 @@
...
 import hashlib
 import json
-import swiftclient
-from swift.common.internal_client import UnexpectedResponse
-from swift.common.utils import FileLikeIter, decode_timestamps
 import sys
 import traceback
 import urllib
 
+import swiftclient
```
[the remark this answers is not in the document]
Comment (thiagodasilva) [reply s3_sync/sync_swift.py]: I'd suggest alphabetical order, but also separating the two groups: `import foo` and `from blah import foo`. wdyt?

Moved `swiftclient` back before `sys`

File: s3_sync/sync_swift.py [hunk 3]
```diff
@@ -50,6 +52,14 @@ def __init__(self, *args, **kwargs):
         self.remote_delete_after_addition = \
             self.settings.get('remote_delete_after_addition',
                               DEFAULT_SEGMENT_DELAY)
+        # if policy is set to expire remote objects, then don't sync
+        # x-delete-after header, the correct header/value will be set
+        # according to policy set by operator
+        self.propagated_headers = set(['content-type'])
+        self.propagate_expiration = config_true_value(
+            self.settings.get('propagate_expiration', 'false'))
```
[the remark this answers is not in the document]
Comment (thiagodasilva) [reply s3_sync/sync_swift.py]: good point, /me keeps referring back to swift-isms...fixing.

File: test/unit/test_sync_swift.py (hunk 1 above)
Comment (thiagodasilva) [reply test/unit/test_sync_swift.py]: fixed

**Comment by timuralp** [comment]
Merged via ecf310d1fc6df75f1beebdda26d070a9b5f36f33. Closing.

**Pull request closed without merging**